query/executor.go

```diff
@@ -289,6 +306,94 @@ func (e *Executor) WithLogger(log *zap.Logger) {
 	e.TaskManager.Logger = e.Logger
 }
 
+func startFileLogWatcher(w WatcherInterface, e *Executor, ctx context.Context) error {
+	path := w.GetLogPath()
+
+	if err := w.Add(path); err != nil {
+		return fmt.Errorf("add watch path: %w", err)
+	}
+
+	for {
+		select {
+		case event, ok := <-w.Event():
+			if !ok {
+				return fmt.Errorf("watcher event channel closed")
+			}
+
+			e.Logger.Debug("received file event", zap.String("event", event.Name))
+
+			if err := func() error {
+				if event.Op == fsnotify.Remove || event.Op == fsnotify.Rename {
+					e.mu.Lock()
+					defer e.mu.Unlock()
+					e.Logger.Info("log file altered, creating new Watcher",
+						zap.String("event", event.Name),
+						zap.String("path", path))
+
+					if err := w.FileChangeCapture(); err != nil {
+						e.mu.Unlock()
+						return fmt.Errorf("handle file change: %w", err)
+					}
+
+					logger := w.GetLogger()
+					e.Logger = logger
+					e.TaskManager.Logger = e.Logger
+				}
+				return nil
+			}(); err != nil {
+				return err
+			}
+
+		case err, ok := <-w.Error():
+			errs := make([]error, 3)
+			errs = append(errs, err)
+
+			closeErr := w.Close()
+			if closeErr != nil {
+				errs = append(errs, fmt.Errorf("failed to close watcher: %w", closeErr))
+			}
+			if !ok {
+				errs = append(errs, fmt.Errorf("watcher error channel closed"))
+			}
+
+			return errors.Join(errs...)
+		case <-ctx.Done():
+			closeErr := w.Close()
+			if closeErr != nil {
+				return fmt.Errorf("failed to close watcher: %w", closeErr)
+			}
+			e.Logger.Info("closing file Watcher")
+			return ctx.Err()
+		}
+	}
+}
+
+func (e *Executor) WithLogWriter(w WatcherInterface, ctx context.Context) {
```

Comment:
`ctx` should be the first parameter. From the [docs](https://pkg.go.dev/context#pkg-overview):

> The Context should be the first parameter, typically named ctx.